From 4119529d8a14e023f0fd22175657208a0ec841a0 Mon Sep 17 00:00:00 2001
From: "Bobby R. Bruce" <bbruce@ucdavis.edu>
Date: Tue, 3 May 2022 17:11:33 -0700
Subject: [PATCH 611/757] util-docker: Fix the broken riscv cross-compiler
 docker

Previously the line
```
RUN git checkout --recurse-submodules 051b9f7ddb7d136777505ea19c70a41926842b96
```

Was causing an error.

Change-Id: Ia13e676f7113c320d4a335b4999c47de415b278b
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/59351
Tested-by: kokoro <noreply+kokoro@google.com>
Maintainer: Jason Lowe-Power <power.jg@gmail.com>
Reviewed-by: Hoa Nguyen <hoanguyen@ucdavis.edu>
---
 util/dockerfiles/llvm-gnu-cross-compiler-riscv64/Dockerfile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/util/dockerfiles/llvm-gnu-cross-compiler-riscv64/Dockerfile b/util/dockerfiles/llvm-gnu-cross-compiler-riscv64/Dockerfile
index f500b9b..df7a58c 100644
--- a/util/dockerfiles/llvm-gnu-cross-compiler-riscv64/Dockerfile
+++ b/util/dockerfiles/llvm-gnu-cross-compiler-riscv64/Dockerfile
@@ -39,9 +39,10 @@ FROM stage1 AS stage2
 RUN mkdir -p /riscv/_install
 WORKDIR /riscv
 ENV PATH=`/riscv/_install/bin:$PATH`
-RUN git clone --recursive https://github.com/riscv/riscv-gnu-toolchain
+RUN git clone https://github.com/riscv/riscv-gnu-toolchain
 WORKDIR /riscv/riscv-gnu-toolchain
-RUN git checkout --recurse-submodules 051b9f7ddb7d136777505ea19c70a41926842b96
+RUN git checkout 051b9f7ddb7d136777505ea19c70a41926842b96
+RUN git submodule update --init --recursive
 RUN ./configure --prefix=/riscv/_install --enable-multilib
 RUN make linux -j`nproc`
 RUN make install
-- 
1.8.3.1

